lib/starknet-jvm.md

```diff
@@ -396,6 +396,109 @@ fun main(args: Array<String>) {
 }
 ```
 
+## Calling multiple contracts: Fetching ETH balance
+
+### In Java
+
+```java
+import com.swmansion.starknet.account.Account;
+import com.swmansion.starknet.account.StandardAccount;
+import com.swmansion.starknet.provider.Provider;
+import com.swmansion.starknet.data.types.*;
+import com.swmansion.starknet.provider.Request;
+import com.swmansion.starknet.provider.rpc.JsonRpcProvider;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        // Create a provider for interacting with Starknet
+        Provider provider = new JsonRpcProvider("https://example-node-url.com/rpc");
```

Comment:
Just making sure, have you tried running this example code? 

It seems to me that this might not work, as `provider` is of `Provider` interface, and `Provider` interface (I assume intentionally) does not support `batchRequests`.

Reply:
Correct, `batchRequests` was is not supported by `Provider`. It is a custom method, and is not an equivalent of real Starknet endpoint. That's why I decided not to include it in `Provider`.

However, it may be potentially confusing for users that they cannot use `batchRequests` on an instance of `Provider`.

Wdyt?

Regarding the test - I've updated sample code and replaced `Provider` with `JsonRpcProvider` 👍

Reply:
> However, it may be potentially confusing for users that they cannot use batchRequests on an instance of Provider.

Reply:
>  It is a custom method and not an equivalent of specified Starknet endpoint. That's why I decided not to include it in Provider.

I would say it's JSON-RPC specific method. Previously, we also had `GatewayProvider` which likely would not support batching. However, currently afaik there are no ways to communicate with Starknet other than JSON-RPC, so theoretically all providers are "JSON-RPC" providers, and the original naming was retained mainly for backward compatibility.

The way I see it, we either:
1. Leave batching only in `JsonRpcProvider`, because it's JSON-RPC specific.
2. Push batching up to `Provider` interface to allow users to keep using the `Provider` as an interface.

It seems to me 2. offers broader utility, however, it slightly breaks abstraction by introducing some underlying request and response mechanics to the provider interface. 

WDYT?

Reply:
While I generally uphold the principle of not violating the level of abstraction, the deviation in this case is rather minimal, making the 2. option possibly acceptable. Consequently, implementing `Provider` interface in `JsonRpcpProvider` currently seems unnecessary and doesn't add significant value. Do we want to keep it, if all providers are "JSON-RPC" providers?

Reply:
>  Do we want to keep it, if all providers are "JSON-RPC" providers?

Discussed offline.

> the deviation in this case is rather minimal, making the 2. option possibly acceptable

The only thing that concerns me is that we'd introduce `HttpRequest` to the method signature of an interface, which is JSON-RPC specific.

I don't have a strong say on what's the best approach here, @THenry14 wdyt?

Reply:
No strong opinion either, but I guess it makes a bit more sense to me to keep it in JsonRpcProvider